### ISSUE TITLE
run web fails: ImportError: No module named amqp

### ISSUE THREAD

**Issue by choffm** [getsentry/docker-sentry#125]
Using the sentry:onbuild image, I am currently unable to deploy sentry on Cloud Foundry.

I am providing a redis and a postgres instance, and a custom docker entrypoint (extended the one provided from this repository) where I export the `SENTRY_DB_*`, `SENTRY_POSTGRES_*` and `SENTRY_REDIS_*` variables before executing sentry run web. I even tried to manually set the redis path in the `BROKER_URL` variable manually. By doing an /usr/bin/env before starting sentry in the entrypoint, I can verify that all required variables are properly setup.

However, sentry startup fails with the following output:

```
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR Traceback (most recent call last):
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/bin/sentry", line 11, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     sys.exit(main())
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 161, in main
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     cli(prog_name=get_prog(), obj={}, max_content_width=100)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return self.main(*args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     rv = self.invoke(ctx)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return _process_result(sub_ctx.command.invoke(sub_ctx))
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return _process_result(sub_ctx.command.invoke(sub_ctx))
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return ctx.invoke(self.callback, **ctx.params)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return callback(*args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return f(get_current_context(), *args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 73, in inner
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return ctx.invoke(f, *args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return callback(*args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     return f(get_current_context(), *args, **kwargs)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 35, in inner
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     configure()
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 124, in configure
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     configure(ctx, py, yaml, skip_service_validation)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/settings.py", line 152, in configure
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     skip_service_validation=skip_service_validation
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/initializer.py", line 260, in initialize_app
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     bootstrap_options(settings, config['options'])
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/initializer.py", line 133, in bootstrap_options
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from sentry.options import load_defaults
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/options/__init__.py", line 10, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from .store import OptionsStore
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/options/store.py", line 19, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from sentry.db.models.query import create_or_update
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/db/models/__init__.py", line 11, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from .base import *  # NOQA
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/db/models/base.py", line 14, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db import models
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/__init__.py", line 5, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models.query import Q
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 17, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models.deletion import Collector
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/deletion.py", line 4, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models import signals, sql
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/__init__.py", line 4, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models.sql.subqueries import *
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/subqueries.py", line 12, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models.sql.query import Query
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 22, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from django.db.models.sql import aggregates as base_aggregates_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/aggregates.py", line 9, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     ordinal_aggregate_field = IntegerField()
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/db/models/fields/__init__.py", line 116, in __init__
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 54, in __getattr__
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     self._setup(name)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 49, in _setup
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     self._wrapped = Settings(settings_module)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 128, in __init__
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     mod = importlib.import_module(self.SETTINGS_MODULE)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/utils/importlib.py", line 40, in import_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     __import__(name)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/importer.py", line 52, in load_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     reraise_as(ConfigurationError(msg))
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/utils/settings.py", line 48, in reraise_as
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     six.reraise(new_type, new_exception, e_traceback)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/importer.py", line 44, in load_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     mod = self._load_module(fullname)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/runner/importer.py", line 63, in _load_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     default_settings_mod = import_module(self.default_settings)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/django/utils/importlib.py", line 40, in import_module
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     __import__(name)
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/sentry/conf/server.py", line 401, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from kombu import Exchange, Queue
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/kombu/__init__.py", line 67, in __getattr__
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     module = __import__(object_origins[name], None, None, [name])
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/kombu/entity.py", line 10, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from .abstract import MaybeChannelBound, Object
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/kombu/abstract.py", line 12, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from .connection import maybe_channel
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/kombu/connection.py", line 19, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from kombu import exceptions
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR   File "/usr/local/lib/python2.7/site-packages/kombu/exceptions.py", line 12, in 
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR     from amqp import ChannelError, ConnectionError, ResourceError
   2018-02-23T15:31:44.66+0100 [APP/PROC/WEB/0] ERR sentry.runner.importer.ConfigurationError: ImportError: No module named amqp
```

However, amqp is installed, as adding a requirements.txt with amqp (and, to go sure, librabbitmq) gives the following output:
```
Step 1/3 : FROM sentry:onbuild
# Executing 4 build triggers...
Step 1/1 : COPY . /usr/src/sentry
Step 1/1 : RUN if [ -s requirements.txt ]; then pip install -r requirements.txt; fi
 ---> Running in d32bb66265ff
Requirement already satisfied: librabbitmq in /usr/local/lib/python2.7/site-packages (from -r requirements.txt (line 1))
Requirement already satisfied: amqp in /usr/local/lib/python2.7/site-packages (from -r requirements.txt (line 2))
```

When doing a `pip freeze` before starting sentry, I also see that `amqp==1.4.9`is installed. Full output:
```
 amqp==1.4.9
 anyjson==0.3.3
 asn1crypto==0.24.0
 BeautifulSoup==3.2.1
 billiard==3.3.0.23
 boto3==1.4.5
 botocore==1.5.70
 cached-property==1.3.1
 celery==3.1.18
 certifi==2018.1.18
 cffi==1.11.4
 chardet==3.0.4
 click==6.7
 contextlib2==0.5.5
 cryptography==2.1.4
 cssselect==1.0.3
 cssutils==0.9.10
 Django==1.6.11
 django-crispy-forms==1.4.0
 django-jsonfield==0.9.13
 django-picklefield==0.3.2
 django-sudo==2.1.0
 django-templatetag-sugar==1.0
 djangorestframework==2.4.8
 docutils==0.14
 email-reply-parser==0.2.0
 enum34==1.1.6
 exam==0.10.6
 futures==3.2.0
 hiredis==0.1.6
 honcho==0.7.1
 httplib2==0.10.3
 idna==2.6
 ipaddress==1.0.19
 jmespath==0.9.3
 kombu==3.0.35
 librabbitmq==1.6.1
 loremipsum==1.0.5
 lxml==4.1.1
 milksnake==0.1.1
 mistune==0.8.3
 mmh3==2.3.1
 mock==1.0.1
 oauth2==1.9.0.post1
 oauthlib==2.0.6
 olefile==0.45.1
 percy==1.0.8
 petname==2.0
 Pillow==4.2.1
 progressbar2==3.10.1
 psycopg2==2.6.2
 py==1.5.2
 pycparser==2.18
 PyJWT==1.5.3
 pyOpenSSL==17.5.0
 pytest==3.1.3
 pytest-django==2.9.1
 pytest-html==1.9.0
 python-dateutil==2.6.1
 python-memcached==1.59
 python-openid==2.2.5
 python-u2flib-server==4.0.1
 python-utils==2.3.0
 pytz==2018.3
 PyYAML==3.11
 qrcode==5.3
 raven==5.33.0
 rb==1.7
 redis==2.10.5
```

Do you have any idea what could be wrong?

**Comment by DamianLion** [comment]
I encounterd the same issue with the non onbuild image

**Comment by BYK** [comment]
This seems to be working over at getsentry/onpremise as of now so closing the issue. If you are still having this problem or something similar, please file a new issue with as many details as you can for your particular setup.
